packages/dynamite/dynamite/lib/src/models/type_result/type_result.dart

```diff
@@ -87,6 +87,11 @@ sealed class TypeResult {
         return 'json.encode($serialized)';
       case 'application/x-www-form-urlencoded':
         return 'Uri(queryParameters: $serialized! as Map<String, dynamic>).query';
+      case 'application/octet-stream':
+        if (className != 'Uint8List') {
+          throw Exception('octet-stream can only be applied to binary data. Expected Uint8List but got $className');
+        }
```

Comment:
Is this check actually necessary?

Reply:
This method is a generic implementation for all TypeResults (unless overridden).

It's just some safety mechanism.